src/display/annotation_layer.js

```diff
@@ -193,20 +191,44 @@ class AnnotationElement {
    *
    * @private
    * @param {boolean} ignoreBorder
+   * @param {boolean} useRect
    * @memberof AnnotationElement
    * @returns {HTMLElement} A section element.
    */
-  _createContainer(ignoreBorder = false) {
-    const { data, page, viewport } = this;
+  _createContainer(ignoreBorder, useRect) {
+    const {
+      data,
+      parent: { page, viewport },
+    } = this;
 
     const container = document.createElement("section");
     container.setAttribute("data-annotation-id", data.id);
 
+    // The accessibility manager will move the annotation in the DOM in
+    // order to match the visual ordering.
+    // But if an annotation is above an other one, then we must draw it
+    // after the other one whatever the order is in the DOM, hence the
+    // use of the z-index.
+    container.style.zIndex = this.parent.zIndex++;
+
+    if (this.data.popupRef) {
+      container.setAttribute("aria-haspopup", true);
+    }
+
     if (data.noRotate) {
       container.classList.add("norotate");
     }
 
     const { pageWidth, pageHeight, pageX, pageY } = viewport.rawDims;
+
+    if (!data.rect || !useRect) {
```

Comment:
Looking at the code it *appears* that `useRect === false` *only* for `PopupAnnotationElement`, unless I'm completely missing something.
Hence I wonder if we could perhaps avoid adding that new parameter and simply do the following instead?
```suggestion
    if (!data.rect || this instanceof PopupAnnotationElement) {
```